File: Qt e Progetti/Progetti/2017-2018 Progetto Calcolatrice/Polygonator - Menon e Dalla Via/C++/Model/Polygons/Quadrilateral.cpp
#include "Quadrilateral.h"

Quadrilateral::Quadrilateral(const list<Point> &vertices) : Polygon(vertices) {}

bool Quadrilateral::canBeCreatedWith(const Polygon &polygon) {
    return polygon.getVertices().size() == 4;
}

string Quadrilateral::getName() const {
    return "Quadrilateral";
}

File: Qt e Progetti/Progetti/2017-2018 Progetto Calcolatrice/Polygonator - Menon e Dalla Via/Java/src/Model/Polygons/Quadrilateral.java
package Model.Polygons;

import Model.Line;
import Model.Point;

import java.util.List;

public class Quadrilateral extends Polygon {
    public Quadrilateral(final List<Point> vertices) throws Exception {
        super(vertices);
    }

    /**
     * Returns whether a list of points can make a quadrilateral or not
     */
    public static boolean canBeCreatedWith(final Polygon polygon) throws Exception {
        return polygon.getVertices().size() == 4;
    }

    @Override
    public String toString() {
        return "Quadrilateral";
    }
}
